src/construction.cpp

```diff
@@ -615,13 +632,21 @@ bool player_can_build( player &p, const inventory &pinv, const std::string &desc
     return false;
 }
 
+bool character_has_skill_for( const Character &c, const construction &con )
```

Comment:
Non-issue: crafting (recipes) also use a `required_skills`, although for a different containing object, and not as cleanly as here: it's still done in `player::exceeds_recipe_requirements( const recipe &rec )`.

This code and that look similar, although this one seems to me more readable, and isn't class-tied.

Extracting this function for general use should be considered for future refactoring - add a `TODO`?.. Relevant to goals outlined in #20220 (although not exactly what you meant there), and Entity-Component-System touched on in #20069.